File: src/Web/WebSPA/Client/src/store/signal/synchronization/synchronized-object-ids.ts
import { Permissions } from 'src/core-hub.types';
import { ParticipantData } from 'src/features/conference/types';
import { SceneOptions } from 'src/features/create-conference/types';
import { EquipmentConnection } from 'src/features/media/types';

export const ROOMS = 'rooms';
export const CONFERENCE = 'conference';
export const PARTICIPANTS = 'participants';
export const PARTICIPANT_PERMISSIONS = 'participantPermissions';
export const CHAT = 'chat';
export const SUBSCRIPTIONS = 'subscriptions';
export const BREAKOUT_ROOMS = 'breakoutRooms';
export const MEDIA = 'media';
export const EQUIPMENT = 'equipment';
export const SCENE = 'scene';
export const SCENE_TALKINGSTICK = 'scene_talkingStick';
export const TEMPORARY_PERMISSIONS = 'temporaryPermissions';

export type SynchronizedParticipants = {
   participants: { [participantId: string]: ParticipantData };
};

export type SynchronizedConferenceInfo = {
   isOpen: boolean;
   moderators: string[];
   scheduledDate?: string | null;
   name: string | null;
   isPrivateChatEnabled: boolean;
   sceneOptions: SceneOptions;
};

export type SynchronizedParticipantsPermissions = {
   permissions: Permissions;
};

export type SynchronizedRooms = {
   rooms: Room[];
   defaultRoomId: string;
   participants: { [participanId: string]: string };
};

export type Room = {
   roomId: string;
   displayName: string;
};

export type ChatSynchronizedObject = {
   participantsTyping: { [id: string]: boolean };
};

export type SynchronizedEquipment = {
   connections: Record<string, EquipmentConnection>;
};

File: src/Services/ConferenceManagement/Strive.Core/Services/SynchronizedObjectIds.cs
namespace Strive.Core.Services
{
    public static class SynchronizedObjectIds
    {
        // global/for all
        public const string ROOMS = "rooms";
        public const string CONFERENCE = "conference";
        public const string PARTICIPANTS = "participants";
        public const string CHAT = "chat";
        public const string BREAKOUT_ROOMS = "breakoutRooms";
        public const string TEMPORARY_PERMISSIONS = "temporaryPermissions";
        public const string POLL = "poll";
        public const string POLL_RESULT = "poll_result";
        public const string POLL_ANSWERS = "poll_answers";

        // individual
        public const string MEDIA = "media";
        public const string EQUIPMENT = "equipment";
        public const string SUBSCRIPTIONS = "subscriptions";
        public const string PARTICIPANT_PERMISSIONS = "participantPermissions";

        // room
        public const string SCENE = "scene";
        public const string SCENE_TALKINGSTICK = "scene_talkingStick";
        public const string WHITEBOARDS = "whiteboards";
    }
}
